### PR TITLE
Tests for DnsZone functions

### PR DESCRIPTION
**Pull request by jcoglan**
This is James from @neighbourhoodie, I have been working on the STF milestone M1 to improve test coverage, focussing on `resolved`. We have written a lot of tests for this module and we are splitting the work into a number of PRs, one for each test file we've added. Each new test file usually corresponds to a single source file, and exercises the functions in that source file directly.

This PR includes tests for `resolved-dns-zone.c`, providing coverage of functions relating to `DnsZone` object.

### CODE REVIEW

**Comment by jcoglan** [comment]
I believe I have fixed all the issues with the files in this PR -- asan/ubsan problems, gcc/clang differences, linter checks, etc. and the remaining failing CI tasks are failing for reasons other than the changes here. If that's incorrect and there are other things I still need to change, do let me know what else I need to do.

File: src/resolve/test-dns-zone.c [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+#include "resolved-dns-scope.h"
+#include "resolved-dns-zone.h"
+#include "resolved-link.h"
+#include "resolved-manager.h"
+
+#include "log.h"
+#include "tests.h"
+
+static void dns_scope_freep(DnsScope **s) {
+        dns_scope_free(*s);
```
Comment (bluca) [inline src/resolve/test-dns-zone.c]: assert or exit early if null

File: src/resolve/test-dns-zone.c [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        Manager manager = {};
+        _cleanup_(dns_scope_freep) DnsScope *scope = NULL;
+        DnsZone *zone = NULL;
+        DnsZoneItem *item = NULL;
+        _cleanup_(dns_resource_record_unrefp) DnsResourceRecord *rr = NULL;
+
+        ASSERT_OK(dns_scope_new(&manager, &scope, NULL, DNS_PROTOCOL_DNS, AF_INET));
+        zone = &scope->zone;
+
+        rr = dns_resource_record_new_full(DNS_CLASS_IN, DNS_TYPE_A, "www.example.com");
```
Comment (bluca) [inline src/resolve/test-dns-zone.c:33]: assert here and other calls below

File: src/resolve/test-dns-zone.c [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+        ASSERT_EQ((int)item->state, DNS_ZONE_ITEM_ESTABLISHED);
+}
+
+TEST(dns_zone_put_any_class_is_invalid) {
+        Manager manager = {};
+        _cleanup_(dns_scope_freep) DnsScope *scope = NULL;
+        DnsZone *zone = NULL;
+        _cleanup_(dns_resource_record_unrefp) DnsResourceRecord *rr = NULL;
+
+        dns_scope_new(&manager, &scope, NULL, DNS_PROTOCOL_DNS, AF_INET);
```
Comment (bluca) [inline src/resolve/test-dns-zone.c:53]: assert here and other calls below

**Comment by jcoglan** [comment]
@bluca I have made the requested changes and put null checks after anything that allocates.

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
merged manually to solve trivial merge conflict

**Pull request closed without merging**